src/Workspaces/Core/Portable/TemporaryStorage/TemporaryStorageServiceFactory.cs

```diff
@@ -199,6 +203,18 @@ public TemporaryTextStorage(TemporaryStorageService service, string storageName,
                 public string? Name => _memoryMappedInfo?.Name;
                 public long Offset => _memoryMappedInfo!.Offset;
                 public long Size => _memoryMappedInfo!.Size;
+                public SourceHashAlgorithm ChecksumAlgorithm => _checksumAlgorithm;
+                public Encoding? Encoding => _encoding;
+
+                public ImmutableArray<byte> GetChecksum()
+                {
+                    if (_checksum.IsDefault)
+                    {
+                        _checksum = ReadText(CancellationToken.None).GetChecksum();
```

Comment:
should we do an interlocked swap here?  that way we always return the exact same array out? it feels a bit strange that i could call and normally get the same result come out, but occasionally get a different array (And that would be obersvable).

Reply:
Don't think it makes a difference but will switch it.

Reply:
➡️ Done